### ISSUE TITLE
Don't install pytest_devpi_server by default?

### ISSUE THREAD

**Issue by The-Compiler** [devpi/devpi#780]
I have a local devpi server instance running on my laptop, as per [running a pypi mirror on your laptop](https://devpi.net/docs/devpi/devpi/stable/+d/quickstart-pypimirror.html).

However, since `server/setup.py` contains:

```python
entry_points={
  # ...
  'pytest11': [
      "pytest_devpi_server = pytest_devpi_server"]}
```

my pytest environment also has the `devpi_server` plugin installed. Thus, when I run `pytest --help` I see things like:

```
custom options:
  --slow                run slow tests involving remote services (pypi.org)
  --backend=BACKEND     run tests with specified dotted name backend
```

or in `--fixtures`:

```
--------------------------- fixtures defined from pytest_devpi_server ---------------------------
devpiserver_makepypistage
    /usr/lib/python3.8/site-packages/pytest_devpi_server/__init__.py:16: no docstring available
```

As a user of devpi (and pytest), I'm not interested in those. Even worse, options like `--slow` and `--backend` might conflict with project's testsuites which then would break as soon as devpi-server is installed on my machine.

It seems to me like that plugin should only be installed when running devpi-server's tests, e.g. via its `tox.ini` or perhaps a `devpi-server[dev]` extra in `setup.py` or so.

Somewhat related: #505

**Comment by fschulze** [comment]
I don't see how I can move that package into extras. I only know how to add extra dependencies, but we don't want to make the tests/plugin a separate release. Do you have an example ``setup.py`` where a package is separated in the same release?

**Comment by The-Compiler** [comment]
Ah, right, I'm not sure whether that's possible.

Is there a reason for it being a globally available plugin, though? Given that it's only used in the tests of devpi-server as much as I can tell, couldn't it just be a `conftest.py`?

**Comment by fschulze** [comment]
It is used for devpi-postgresql and I think devpi-client, but I didn't check the last one. I would love to find a sane solution that works with actual releases and development installations.
But wouldn't it be better to install devpi-server outside of your path in a virtualenv? Only installing devpi-client wouldn't cause this issue.

**Comment by The-Compiler** [comment]
I [package](https://aur.archlinux.org/packages/devpi-server/) devpi-server as an Archlinux package, thus I have it installed system-wide. I'd rather not set up virtualenvs for projects I'm not developing on, especially when they run system-wide with systemd-units and all.

I suppose I could just patch out the plugin in the AUR package, but that seems a bit of a hack as well.

**Comment by fschulze** [comment]
It seems like the entrypoint for pytest could be made optional with an ``extras``: https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras-optional-features-with-their-own-dependencies

**Comment by The-Compiler** [comment]
Sorry for the late answer, this got buried in my inbox. I wasn't aware that it's possible to tie entry points to extra dependencies. It's a bit unclear to me how it works in detail (and if it works with how pytest uses entrypoints), but I suppose it'd be worth a try. At the moment I can't commit to a contribution from my side, though.

**Comment by fschulze** [comment]
I tried the described stuff from the setuptools docs and they don't seem to work as described. The extras declaration is just appended to ``devpi_server.egg-info/entry_points.txt`` instead of being skipped.

I don't see a way out of this. Importing from another conftest doesn't work correctly and is deprecated.

What I need is a way for devpi-postgresql to run the devpi-server tests of the installed devpi-server version and use its fixtures (as their implementation can change between versions).

Basically I need "extras" to work for any package option, then there could be a ``tests`` extra which includes the test data and the pytest plugin, but they won't be installed without it.

**Comment by The-Compiler** [comment]
And again I'm awfully late with my reply :see_no_evil: 

No idea what to do about it then, either (other than having a separate `devpi-testutils` release, but I can see how that's rather cumbersome). The only thing I could probably do is patching this out in the Archlinux package, but that just strikes me as odd as well...

Maybe @nicoddemus has some good idea how to handle this?

**Comment by fschulze** [comment]
I just learned about ``pytest_plugins``, which might be the solution. In my testing pytest still picks up ``pytest_devpi_server``, even if I remove it from the entry points, but that might be related to my setup. Can you confirm that if you remove the entry point your pytest doesn't pick up the plugin anymore?
